File: src/main/kotlin/tech/diggle/apps/qikpay/payments/DateTo8601.java
package tech.diggle.apps.qikpay.payments;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.util.StdConverter;

import java.io.IOException;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;
import java.util.Date;

class DateTo86013 {

    @JsonSerialize(converter = DateTo8601.class)
    Date createdAtl;
}

File: src/main/kotlin/tech/diggle/apps/qikpay/DateTo8601.kt
package tech.diggle.apps.qikpay

import com.fasterxml.jackson.databind.util.StdConverter
import java.time.LocalDateTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.*

class DateTo8601 : StdConverter<Date, String>() {
    override fun convert(date: Date): String {
        return LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault()).format(DateTimeFormatter.ISO_DATE_TIME)
    }
}